optaplanner-benchmark/src/test/java/org/optaplanner/benchmark/config/PlannerBenchmarkConfigTest.java

```diff
@@ -63,4 +69,100 @@ public void invalidNameWithPrefixWhitespace() {
         config.validate();
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void noSolverConfigs() {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        config.setSolverBenchmarkConfigList(null);
+        config.setSolverBenchmarkBluePrintConfigList(null);
+        config.validate();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void nonUniqueSolverConfigName() {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        final String sbcName = "x";
+        SolverBenchmarkConfig sbc1 = new SolverBenchmarkConfig();
+        sbc1.setName(sbcName);
+        SolverBenchmarkConfig sbc2 = new SolverBenchmarkConfig();
+        sbc2.setName(sbcName);
+        config.setSolverBenchmarkConfigList(Arrays.asList(sbc1, sbc2));
+        config.generateSolverBenchmarkConfigNames();
+    }
+
+    @Test
+    public void uniqueNamesGenerated() {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        SolverBenchmarkConfig sbc1 = new SolverBenchmarkConfig();
+        SolverBenchmarkConfig sbc2 = new SolverBenchmarkConfig();
+        SolverBenchmarkConfig sbc3 = new SolverBenchmarkConfig();
+        sbc3.setName("Config_1");
+        List<SolverBenchmarkConfig> configs = Arrays.asList(sbc1, sbc2, sbc3);
+        config.setSolverBenchmarkConfigList(configs);
+        config.generateSolverBenchmarkConfigNames();
+        assertEquals("Config_1", sbc3.getName());
+        TreeSet<String> names = new TreeSet<String>();
+        for (SolverBenchmarkConfig sc : configs) {
+            names.add(sc.getName());
+        }
+        for (int i = 0; i < configs.size(); i++) {
+            assertTrue(names.contains("Config_" + i));
+        }
+    }
+
+    @Test
+    public void resolveParallelBenchmarkCountAutomatically() {
+        PlannerBenchmarkConfig config = new PlannerBenchmarkConfig();
+        assertEquals(-1, config.resolveParallelBenchmarkCountAutomatically(-1));
+        assertEquals(0, config.resolveParallelBenchmarkCountAutomatically(0));
```

Comment:
Good point. There is always at least 1 available processor even if `availableProcessors()` returned 0 or less, otherwise the code wouldn't be running :smile: